src/ansys/edb/core/geometry/polygon_data.py

```diff
@@ -434,11 +444,12 @@ def intersection_type(self, other, tol=1e-9):
 
         Returns
         -------
-        bool
-            ``True`` when successful, ``False`` when failed.
+        IntersectionType
         """
-        return self.__stub.GetIntersectionType(
-            messages.polygon_data_pair_with_tolerance_message(self, other, tol)
+        return IntersectionType(
+            self.__stub.GetIntersectionType(
+                messages.polygon_data_pair_with_tolerance_message(self, other, tol)
+            )
```

Comment:
Should this be 
```
return IntersectionType(
    self.__stub.GetIntersectionType(
        messages.polygon_data_pair_with_tolerance_message(self, other, tol)
    ).intersection_type
)
```

Reply:
@chenchienjacklin Indeed it should be! Thanks for catching that. I somehow missed this when testing my changes. This should be fixed now.